force-app/main/default/lwc/libsMomentjs/__tests__/libsMomentjs.test.js

```diff
@@ -0,0 +1,116 @@
+                return new Promise((resolve, reject) => {
+                    global.moment = require('../../../staticresources/moment');
+                    resolve();
+                });
+            }
+        };
+    },
+    { virtual: true }
+);
+
+// Sample error for loadScript error
+const LOAD_SCRIPT_ERROR = {
+    body: { message: 'An internal server error has occurred' },
+    ok: false,
+    status: 400,
+    statusText: 'Bad Request'
+};
+
+describe('c-libs-momentjs', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+    });
+
+    // Helper function to wait until the microtask queue is empty. This is needed for promise
+    // timing when the platformResourceLoader promises.
+    function flushPromises() {
+        // eslint-disable-next-line no-undef
+        return new Promise(resolve => setImmediate(resolve));
+    }
+
+    it('loads the momentjs javascript static resource', () => {
+        const MOMENT_JS = 'moment';
+
+        // Create initial element
+        const element = createElement('c-libs-momentjs', {
+            is: LibsMomentjs
+        });
+        document.body.appendChild(element);
+
+        // Validation that the loadScript promise is called once.
+        // expect(loadScript).resolves.toBe(true);
+        // Validation that the chartjs static resource is passed as parameter.
+        return flushPromises().then(() => {
+            expect(loadScript.mock.calls[0][1]).toEqual(MOMENT_JS);
+        });
+    });
+
+    it('populates the disabled lightning-input fields with moment.js data based on user input', () => {
+        const INPUT_RAW = '2019-03-11T22:30:00.000Z';
+        const OUTPUT_EXPECTED = [70, 11];
+
+        // Create initial element
+        const element = createElement('c-libs-momentjs', {
+            is: LibsMomentjs
+        });
+        document.body.appendChild(element);
+
+        // Selecting the input element for simulating user input
+        const inputEl = element.shadowRoot.querySelector('lightning-input');
+        inputEl.value = new Date(INPUT_RAW);
+        inputEl.dispatchEvent(new CustomEvent('change'));
+
+        return Promise.resolve().then(() => {
+            // Querying all lightning-input fields, and populating the const
+            // based on the disabled property on each lightning-input field.
+            const values = Array.from(
+                element.shadowRoot.querySelectorAll('lightning-input')
+            )
+                .filter(input => input.disabled)
+                .splice(0, 2)
+                .map(input => input.value);
+            expect(values).toEqual(OUTPUT_EXPECTED);
+        });
+    });
+
+    it('shows the error panel element on static resource load error', () => {
+        loadScript.mockRejectedValue(LOAD_SCRIPT_ERROR);
```

Comment:
Same issue here. Since this is not a `jest.fn()` anymore, we don't have the `mockRejectedValue` API available. 

One technique you could try here is to set a variable in the test case before creating the element under test and then having the `loadScript()` mock fork on the value of that variable to decide whether to load the actual library or throw an error.

Reply:
Great recommendation - implemented!